lcompilers/lpython
# add flag `–show-errors` for lsp error highlight

This PR -

1. Replaces the existing LSP implementation with [Microsoft’’s official language server module](https://github.com/microsoft/vscode-languageserver-node).
2. Adds a flag --show-errors which prints the diagnostics from the file to the console. This is used by VSCode Language Server extension to catch the stdout and parse into JSON,

 Steps to test it - 

1. Clone https://github.com/ankitaS11/lpython and check out the` lsp/add_flag/show-error` branch.
2. Compile lpython with LSP flag on.

``` 
conda activate lp # or use your environment name here
cmake -DCMAKE_BUILD_TYPE=Debug -DWITH_LLVM=yes -DWITH_STACKTRACE=yes -DWITH_LFORTRAN_BINARY_MODFILES=no -DWITH_LSP=yes .
cmake --build . -j16
```

3. To get the errors/ warnings (if any) for the given file path in JSON format: run `./src/bin/lpython --show-errors <file_name>` in terminal. This will be used by LPython’s VSCode extension for the error highlight feature.

To Test it in VSCode extension -  

Clone LPython’s VSCode Extension: https://github.com/ankitaS11/lpyth and follow the instructions mentioned in its README.
